lib/pat.c

```diff
@@ -5262,15 +5269,20 @@ grn_pat_inspect_nodes(grn_ctx *ctx, grn_pat *pat, grn_obj *buf)
 {
   pat_node *node;
   grn_obj key_buf;
+  bool is_large_mode = pat_key_is_large_total_key_size(pat);
 
   GRN_TEXT_PUTS(ctx, buf, "{");
   PAT_AT(pat, GRN_ID_NIL, node);
-  if (node->lr[1] != GRN_ID_NIL) {
-    GRN_TEXT_PUTS(ctx, buf, "\n");
-    GRN_OBJ_INIT(&key_buf, GRN_BULK, 0, pat->obj.header.domain);
-    grn_pat_inspect_node(ctx, pat, node->lr[1], -1, &key_buf, 0, "", buf);
-    GRN_OBJ_FIN(ctx, &key_buf);
-    GRN_TEXT_PUTS(ctx, buf, "\n");
+  if (is_large_mode) {
+    ; // TODO: implement if pat node have large key
+  } else {
+    if (node->lr[1] != GRN_ID_NIL) {
+      GRN_TEXT_PUTS(ctx, buf, "\n");
+      GRN_OBJ_INIT(&key_buf, GRN_BULK, 0, pat->obj.header.domain);
+      grn_pat_inspect_node(ctx, pat, node->lr[1], -1, &key_buf, 0, "", buf);
+      GRN_OBJ_FIN(ctx, &key_buf);
+      GRN_TEXT_PUTS(ctx, buf, "\n");
+    }
```

Comment:
Do we need this change in this PR...?

Reply:
I think also that this change is needless in this PR. However, when unused function exist, clang raise the following error. 

```
/home/runner/work/groonga/groonga/lib/pat.c:97:1: error: unused function 'pat_key_is_large_total_key_size' [-Werror,-Wunused-function]
   97 | pat_key_is_large_total_key_size(grn_pat *pat)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```
https://github.com/groonga/groonga/actions/runs/15344296603/job/43177061984#step:8:397 

So, I added this change in this PR.

Reply:
Should we add `-Wno-unused-function` as below?

```diff
modified   CMakeLists.txt
@@ -357,6 +357,7 @@ endif()
 if(GRN_C_COMPILER_GNU_LIKE)
   check_build_flag("-Wall")
   check_build_flag("-Wno-unused-but-set-variable")
+  check_build_flag("-Wno-unused-function")
   check_build_flag("-Wno-unused-parameter")
   check_cflag("-Wno-pointer-sign")
   check_build_flag("-Wfloat-equal")
```

Reply:
Can we implement a small feature that use `pat_is_key_large()`?

Reply:
I see. I'll try it.

Reply:
I implemented a small feature that use `pat_is_key_large()`.